grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -349,11 +351,35 @@ class QueryService implements Status {
                 .setString('consentKey', consentKey)
                 .list()
         results.collect {
-            sampleInfo.put(it, StorageDocument.findAllByConsentCollectionLinkId(it.id))
+            consentCollectionIds.add(it.id)
+            consentCollectionLinkList.add(it)
+        }
+        List <StorageDocument> storageDocuments = findAllDocumentsBySampleCollectionId(consentCollectionIds)
+        consentCollectionLinkList.collect {
```

Comment:
I think you want an `each` here. `collect` returns a collection of items that you're not really using here.